src/main/java/org/fogbowcloud/manager/core/plugins/compute/openstack/OpenStackNovaV2ComputePlugin.java

```diff
@@ -361,15 +365,54 @@ protected String doGetRequest(String endpoint, Token localToken) throws RequestE
 
     @Override
     public ComputeOrderInstance getInstance(Token localToken, String instanceId) throws RequestException {
-        return null;
+        LOGGER.info("Getting instance " + instanceId + " with token " + localToken);
+
+        String tenantId = getTenantId(localToken);
+        String requestEndpoint = getComputeEndpoint(tenantId, SERVERS + "/" + instanceId);
+
+        String jsonResponse = doGetRequest(requestEndpoint, localToken);
+
+        LOGGER.debug("Getting instance from json: " + jsonResponse);
+        return getInstanceFromJson(jsonResponse);
     }
 
-    @Override
-    public List<ComputeOrderInstance> getInstances(Token localToken) throws RequestException {
-        return null;
+    private ComputeOrderInstance getInstanceFromJson(String jsonResponse) throws RequestException {
+        try {
+            JSONObject rootServer = new JSONObject(jsonResponse);
+            JSONObject serverJson = rootServer.getJSONObject(SERVER_JSON_FIELD);
+
+            String id = serverJson.getString(ID_JSON_FIELD);
+            InstanceState state = getInstanceState(serverJson.getString(STATUS_JSON_FIELD));
+
+            return new ComputeOrderInstance(id, state);
+        } catch (JSONException e) {
+            LOGGER.warn("There was an exception while getting instances from json.", e);
+            throw new RequestException();
+        }
     }
 
-    private List<ComputeOrderInstance> getInstancesFromJson(String json) {
+    /**
+     * This method will map Openstack instance status to Fogbow instance status.
+     *
+     * @param instanceStatus status from JSON.
+     * @return {@link InstanceState}
+     */
+    private InstanceState getInstanceState(String instanceStatus) {
+        switch (instanceStatus.toLowerCase()) {
+            case ACTIVE_STATUS:
+                return InstanceState.ACTIVE;
+
+            case ERROR_STATUS:
+                return InstanceState.FAILED;
+
+            default:
+                return InstanceState.INACTIVE;
+        }
+    }
+
+    // TODO: Is it necessary?
```

Comment:
was anyone using it on the old code base?

Reply:
I don't think so.